### PR TITLE
Allow "content" after headings to include that inside an iframe

### PR DESCRIPTION
**Pull request by andymsuk**
Fixes https://github.com/bbc/bbc-a11y/issues/238

### CODE REVIEW

File: features/standards/mag/structure/02_headings.feature [hunk 1]
```diff
@@ -280,3 +280,62 @@ Feature: Headings
       """
     When I test the "Structure: Headings: Content must follow headings" standard
     Then it passes
+
+  @html @automated
```
Comment (joshski) [inline features/standards/mag/structure/02_headings.feature]: These scenarios look like they are indented inconsistently with the rest of the file

Comment (andymsuk) [reply features/standards/mag/structure/02_headings.feature]: Think it's the same now?

File: lib/standards/tests/contentMustFollowHeadings.js [hunk 2]
```diff
@@ -15,11 +15,17 @@ module.exports = {
...
+      return $node.text().trim().length > 0
+    }
+
     $(headingSelector).each(function (index, heading) {
       var nextNode = heading.nextSibling || ancestorsNextSibling(heading)
       while (nextNode) {
-        if ($(nextNode).is(heading.tagName)) { break }
-        if ($(nextNode).text().trim().length > 0) { return }
+        var $nextNode = $(nextNode)
+        var $nextIframeNodesBody = $(nextNode).find('iframe:visible').contents().find('body')
```
Comment (ChrisBAshton) [inline lib/standards/tests/contentMustFollowHeadings.js]: This will find an iframe _wherever_ it is inside the `<div>`, so would this case pass when it should actually fail?

```
<h1>Main heading</h1>
<div>
  <h2>This heading should fail the test</h2>
  <iframe src="perfect.html">...but this iframe will be found to have contents, therefore pass the test</iframe>
</div>
```

File: features/standards/mag/structure/02_headings.feature [hunk 3]
```diff
@@ -280,3 +280,62 @@ Feature: Headings
...
+  Scenario: Heading followed by a hidden iframe with content
+    Given a website running at http://localhost:54321
+    And a page with the body:
+      """
+      <h1>Main heading</h1>
+      <div>
+        <iframe style="display: none" src="http://localhost:54321/perfect.html"></iframe>
+      </div>
```
Comment (ChrisBAshton) [inline features/standards/mag/structure/02_headings.feature]: Can we have a test case for if the `<iframe`> is not inside a `<div>`? I suspect it would fail right now.

File: lib/standards/tests/contentMustFollowHeadings.js (hunk 2 above)
Comment (andymsuk) [reply lib/standards/tests/contentMustFollowHeadings.js]: Why should this fail?

File: features/standards/mag/structure/02_headings.feature (hunk 3 above)
Comment (andymsuk) [reply features/standards/mag/structure/02_headings.feature]: Good spot. I'll add and fix!

File: lib/standards/tests/contentMustFollowHeadings.js (hunk 2 above)
Comment (ChrisBAshton) [reply lib/standards/tests/contentMustFollowHeadings.js]: Isn't this scenario about testing that there is content after each heading? But in my example code above, we have a h1 immediately followed by a h2 without any content. Or have I misunderstood?

Comment (andymsuk) [reply lib/standards/tests/contentMustFollowHeadings.js]: Another heading still counts as "content". See lines 176-185 for an example.

Comment (ChrisBAshton) [reply lib/standards/tests/contentMustFollowHeadings.js]: Oh ok, the "no heading immediately after another heading" rule only applies to headings _of the same level_, I guess?

https://github.com/andymsuk/bbc-a11y/blob/676755ea552dab1f3b9d898d504f24ae064295f3/features/standards/mag/structure/02_headings.feature#L194-L217

* h1 next to h2 passes
* h2 next to h2 fails

...right?

(Just trying to understand the difference.)

Also, ok, let's take my code and change it a bit:

```
<h1>Main heading</h1>
<div>
  <h1>This heading should fail the test</h1>
  <iframe src="perfect.html">...but this iframe will be found to have contents, therefore pass the test</iframe>
</div>
```

...^^ I think this PR might now make this code pass when it shouldn't, because it'll find an iframe and check that?

Comment (andymsuk) [reply lib/standards/tests/contentMustFollowHeadings.js]: This particular standard will pass, because there is content after each of the `h1` elements.
However, the standard for `Exactly one main heading` will fail, as there are now two `h1` elements.

If you change it to this though...
```
      <h1>Main heading</h1>
      <h2>This is a sub-heading</h2>
      <h2>This heading should fail the test</h2>
      <iframe src="http://localhost:54321/perfect.html"></iframe>
```
it does also fail because there's a `h2` after the `h2`.

__However__, if you change it to this...
```
      <h1>Main heading</h1>
      <h2>This is a sub-heading</h2>
      <div>
        <h2>This heading should fail the test</h2>
        <iframe src="http://localhost:54321/perfect.html"></iframe>
      </div>
```
Then it does incorrectly pass 😢 

You're great at finding these edge cases!!

*back to the drawing board*

**Comment by EmmaJP** [comment]
Is this code able to check that the iframe is not empty of content (text/images etc)?

**Comment by andymsuk** [comment]
@EmmaJP It does do that, yh.

**Comment by EmmaJP** [comment]
Have you figured out why it passes? Would it be better to merge this fix and raise an additional issue?

**Comment by andymsuk** [comment]
I've not figured it out yet. And yes - good point. 

This currently passes on `master` branch and shouldn't as there's no content between the two `h2`s:
```
      <h1>Main heading</h1>
      <h2>This is a sub-heading</h2>
      <div>
        <h2>This heading should fail the test</h2>
        <iframe src="http://localhost:54321/perfect.html"></iframe>
      </div>
```

In that case, this PR is ready for approval, from my point of view and I've raised https://github.com/bbc/bbc-a11y/issues/267
